src/main/java/uk/gov/hmcts/divorce/systemupdate/event/SystemNotifyApplicantPartnerNotAppliedForFinalOrder.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.divorce.systemupdate.event;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.divorce.common.notification.PartnerNotAppliedForFinalOrderNotification;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+import uk.gov.hmcts.divorce.notification.NotificationDispatcher;
+
+import static uk.gov.hmcts.ccd.sdk.type.YesOrNo.YES;
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingJointFinalOrder;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SYSTEMUPDATE;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Component
+public class SystemNotifyApplicantPartnerNotAppliedForFinalOrder implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String SYSTEM_PARTNER_NOT_APPLIED_FOR_FINAL_ORDER = "system-notify-applicant-partner-not-applied-final-order";
+
+    @Autowired
+    private NotificationDispatcher notificationDispatcher;
+
+    @Autowired
+    private PartnerNotAppliedForFinalOrderNotification partnerNotAppliedForFinalOrderNotification;
+
+    @Override
+    public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        configBuilder
+            .event(SYSTEM_PARTNER_NOT_APPLIED_FOR_FINAL_ORDER)
+            .forStates(AwaitingJointFinalOrder)
+            .name("Partner not applied for FO")
+            .description("Notify Applicant that partner not applied for Final Order")
+            .grant(CREATE_READ_UPDATE, SYSTEMUPDATE)
+            .grantHistoryOnly(SOLICITOR, CASE_WORKER, SUPER_USER, LEGAL_ADVISOR)
+            .retries(120, 120)
+            .aboutToSubmitCallback(this::aboutToSubmit);
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<CaseData, State> details,
+                                                                       CaseDetails<CaseData, State> beforeDetails) {
+        CaseData data = details.getData();
+
+        notificationDispatcher.send(partnerNotAppliedForFinalOrderNotification, data, details.getId());
```

Comment:
should we move this to submitted callback if case fails to update then notification will be sent which has happened on few occassion